examples/asr/experimental/audio_processing/process_audio.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import glob
+import json
+import os
+from dataclasses import dataclass, is_dataclass
+from pathlib import Path
+from typing import List, Optional
+
+import pytorch_lightning as pl
+import torch
+from omegaconf import OmegaConf
+
+from nemo.collections.asr.models import AudioProcessingModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging, model_utils
+
+
+"""
+Process audio file on a single CPU/GPU. Useful for processing of moderate amounts of audio data.
+
+# Arguments
+    model_path: path to .nemo checkpoint for an AudioProcessingModel
+    pretrained_name: name of a pretrained AudioProcessingModel model (from NGC registry)
+    audio_dir: path to directory with audio files
+    dataset_manifest: path to dataset JSON manifest file (in NeMo format)
+
+    input_channel_selector: list of channels to take from audio files, defaults to `None` and takes all available channels
+    input_key: key for audio filepath in the manifest file, defaults to `audio_filepath`
+
+    output_dir: Directory where processed files will be saved
+    output_filename: Output filename where manifest pointing to processed files will be written
+    batch_size: batch size during inference
+
+    cuda: Optional int to enable or disable execution of model on certain CUDA device.
+    amp: Bool to decide if Automatic Mixed Precision should be used during inference
+    audio_type: Str filetype of the audio. Supported = wav, flac, mp3
+
+    overwrite_output: Bool which when set allowes repeated processing runs to overwrite previous results.
+
+# Usage
+AudioProcessingModel can be specified by either `model_path` or `pretrained_name`.
+Data for processing can be defined with either `audio_dir` or `dataset_naminfest`.
+Processed audio is saved in `output_dir`, and a manifest for processed files is saved
+in `output_filename`.
+
+```
+python process_audio.py \
```

Comment:
can we change the filename to 'inference.py'? everyone understands what is it stands for than 'process.py'